src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -91,6 +97,112 @@ enum WarriorMisc
     SPELL_VISUAL_BLAZING_CHARGE = 26423
 };
 
+// 152278 - Anger Management
+class spell_warr_anger_management_proc : public AuraScript
+{
+    static bool ValidateSpec(ProcEventInfo const& eventInfo, ChrSpecialization spec)
+    {
+        if (Player const* player = eventInfo.GetActor()->ToPlayer())
+            return player->GetPrimarySpecialization() == spec;
+
+        return false;
+    }
+
+    static int32 GetRageSpellCost(SpellInfo const* spellInfo)
+    {
+        for (SpellPowerEntry const* powerCost : spellInfo->PowerCosts)
+            if (powerCost && powerCost->PowerType == POWER_RAGE)
+                return powerCost->ManaCost / 10;
+
+        return 0;
+    }
+
+    static bool CheckArmsProc(AuraEffect const* /*aurEff*/, ProcEventInfo const& eventInfo)
```

Comment:
all these static checks for spec are duplications, you can do instead:

```suggestion
    static auto SpecializationChecker(ChrSpecialization spec)
    {
        return [p_Spec](AuraEffect const* effect, ProcEventInfo const& eventInfo) -> bool
        {
            if (Player const* player = eventInfo.GetActor()->ToPlayer())
                return player->GetPrimarySpecialization() == spec;
            return false;
        };
    }
```

and then durig register:

```suggestion
DoCheckEffectProc += AuraCheckEffectProcFn(SpecializationChecker(ChrSpecialization::WarriorArms), EFFECT_0, SPELL_AURA_DUMMY);
```

and the rest